Exiled.Events/Patches/Events/Player/SearchPickupRequesting.cs

```diff
@@ -0,0 +1,76 @@
+// -----------------------------------------------------------------------
+// <copyright file="SearchPickupRequesting.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Player
+{
+#pragma warning disable SA1313
+    using System;
+
+    using Exiled.API.Features;
+    using Exiled.Events.EventArgs;
+
+    using HarmonyLib;
+
+    using InventorySystem.Searching;
+
+    using Mirror;
+    using Mirror.LiteNetLib4Mirror;
+
+    /// <summary>
+    /// Patches <see cref="SearchCoordinator.ReceiveRequestUnsafe"/>.
+    /// Adds the <see cref="Handlers.Player.SearchPickupRequesting"/> event.
+    /// </summary>
+    [HarmonyPatch(typeof(SearchCoordinator), nameof(SearchCoordinator.ReceiveRequestUnsafe))]
+    internal static class SearchPickupRequesting
+    {
+        private static void Prefix(SearchCoordinator __instance, ref bool __result, out SearchSession? session, out SearchCompletor completor)
+        {
+            try
+            {
+                SearchRequest request = __instance.SessionPipe.Request;
+                completor = SearchCompletor.FromPickup(__instance, request.Target, __instance.ServerMaxRayDistanceSqr);
+                SearchPickupRequestEventArgs ev = new SearchPickupRequestEventArgs(Player.Get(__instance.Hub), request.Target, request.Target.SearchTime, completor.ValidateStart());
```

Comment:
Pass to the constructor all the arguments you can pass, and make the constructor able to accept them

Reply:
i have added the two other value